src/main/java/se/iths/SpringbootGroupProject/services/MessageService.java

```diff
@@ -0,0 +1,46 @@
+package se.iths.SpringbootGroupProject.services;
+
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+import se.iths.SpringbootGroupProject.dto.PublicMessageAndUsername;
+import se.iths.SpringbootGroupProject.repositories.MessageRepository;
+
+import java.util.List;
+
+@Service
+public class MessageService {
+
+    MessageRepository messageRepository;
+
+    public MessageService(MessageRepository messageRepository) {
+        this.messageRepository = messageRepository;
+    }
+
+    @Cacheable("messages")
+    public List<PublicMessageAndUsername> findAllByPrivateMessageIsFalse() {
+        return messageRepository.findAllByPrivateMessageIsFalse()
+                .stream()
+                .map(message -> new PublicMessageAndUsername(
+                        message.getDate(),
+                        message.getTitle(),
+                        message.getMessageBody(),
+                        message.getUser().getUserName())).toList();
+    }
+
+    @CacheEvict(value = "messages", allEntries = true)
+    public void setMessagePrivacy(boolean isPrivate, Long id) {
+        messageRepository.setMessagePrivacy(isPrivate, id);
+    }
+
+    @CacheEvict(value = "messages", allEntries = true)
+    public void editMessage(String updatedBody, Long id) {
+        messageRepository.editMessage(updatedBody, id);
+    }
+
+    @CacheEvict(value = "messages", allEntries = true)
+    public void editTitle(String updatedTitle, Long id) {
+        messageRepository.editTitle(updatedTitle, id);
+    }
```

Comment:
Since these methods are altering our database, maybe they should have the annotation @Transactional? Or is that built in since these are Springboots automatic methods and not our own with our own SQL-query?

Reply:
I was thinking about this. Delving deeper into the topic, I found out that it's good practice to have the `@Transactional` annotation.
https://docs.spring.io/spring-framework/reference/data-access/transaction/declarative/annotations.html
![image](https://github.com/Win-ther/SpringbootGroupProject/assets/143085217/6bbed7f3-0e58-4ea3-9818-ddfd0d96aa5d)

I will get right onto it 👍